cyclonedx/model/bom.py

```diff
@@ -363,6 +364,26 @@ def has_vulnerabilities(self) -> bool:
         """
         return any(c.has_vulnerabilities() for c in self.components)
 
+    def validate(self) -> bool:
+        """
+        Perform data-model level validations to make sure we have some known data integrity prior to attempting output
+        of this `Bom`
+
+        Returns:
+             `bool`
+        """
+
+        # 1. Make sure dependencies are all in this Bom.
+        all_component_bom_refs = set(map(lambda c: c.bom_ref, self.components))
+        all_dependency_bom_refs = set().union(*(c.dependencies for c in self.components))
+        dependency_diff = list(all_dependency_bom_refs.difference(all_component_bom_refs))
```

Comment:
check for the root component of `metadata`

this is the base of the graph.
see https://cyclonedx.org/use-cases/#dependency-graph